### PR TITLE
Test cases for Permission management. 

### PR DESCRIPTION
**Pull request by menakaj**

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml [hunk 1]
```diff
@@ -222,6 +222,18 @@
             <artifactId>testng</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-testng</artifactId>
+            <version>1.7.0</version>
```
Comment (sinthuja) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml]: Can you include this dependency in the root pom? And then import it here in the component pom? With that, we can make sure we reuse the same version of the dependency.

Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml]: ok

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/permission/mgt/PermissionManagerServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@PrepareForTest(PermissionUtils.class)
+public class PermissionManagerServiceTest {
+
+    private Permission permission;
+    private Logger log = Logger.getLogger(PermissionManagerServiceTest.class);
+    private String PERMISSION_URL = "permission/admin/device-mgt/test/testPermission";
+    private String PERMISSION_PATH = "permission/admin/device-mgt/test/testPermission";
+    private String PERMiSSION_METHOD = "ui.execute";
```
Comment (sinthuja) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/permission/mgt/PermissionManagerServiceTest.java]: PERMiSSION_METHOD needs to be PERMISSION_METHOD. Note 'I' not and not i. And also as above variables are constants you can use 'private static final'.

**Review by sinthuja — CHANGES_REQUESTED** [review]
Please fix the comments added.

**Review by Kamidu — APPROVED** [review]

**Review by sinthuja — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml [hunk 3]
```diff
@@ -222,6 +222,14 @@
             <artifactId>testng</artifactId>
         </dependency>
         <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-testng</artifactId>
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml:226]: Shall we make the scope of the test dependencies as "test"

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/permission/mgt/PermissionManagerServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import org.wso2.carbon.device.mgt.common.permission.mgt.PermissionManagementException;
+import org.wso2.carbon.device.mgt.common.permission.mgt.PermissionManagerService;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+import java.util.Properties;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@PrepareForTest(PermissionUtils.class)
+public class PermissionManagerServiceTest {
```
Comment (Megala21) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/permission/mgt/PermissionManagerServiceTest.java:39]: Shall we add a class level comment for the test case

Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/permission/mgt/PermissionManagerServiceTest.java:39]: This is fixed in https://github.com/wso2/carbon-device-mgt/pull/1082/commits/2bff695ea3e455a977bb1a72e146b63f8660c59c

File: components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml (hunk 3 above)
Comment (menakaj) [reply components/device-mgt/org.wso2.carbon.device.mgt.core/pom.xml:226]: This is fixed in https://github.com/wso2/carbon-device-mgt/commit/6bfce09159ce282613c51f4c380f2b6ae0f222ac

**Pull request merged**